docs/dev/types_serialization.md

```diff
@@ -115,6 +115,8 @@ conditions are fulfilled:
   or have a public getter- and a setter- method that follows the Java beans
   naming conventions for getters and setters.
 
+Note that when a data type can't be recognized as a POJO type, it will be handled as GenericType.
```

Comment:
"... it will be processed as a GenericType and serialized with Kryo"? Is it generally understood that use of Kryo can have a significant impact on performance? It's not just serialization which is slower. It looks like if the class does not implement `NormalizedKey` then objects are deserialized for each comparison.

Reply:
Sorry to reply late, and thank you for your suggestions @greghogan 
In fact, I have some doubts about what information to be provided to the users when the data can't be recognized as POJO type. Do you think data type and serialization are sufficient here? thanks